x/session/keeper/keeper.go

```diff
@@ -54,15 +54,15 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
-// BeginBlocker is called at the beginning of every block
+// BeginBlocker is called at the beginning of every block.
 // It fetches the block hash from the committed block ans saves its hash
 // in the store.
 func (k Keeper) BeginBlocker(ctx sdk.Context) {
 	// ctx.BlockHeader().LastBlockId.Hash is the hash of the last block committed
 	hash := ctx.BlockHeader().LastBlockId.Hash
 	// ctx.BlockHeight() is the height of the current block being processed,
 	// decrementing it by 1 gives us the height of the last block committed.
-	height := ctx.BlockHeight() - 1
+	height := ctx.BlockHeader().Height - 1
```

Comment:
Do we still require the -1 here?

Reply:
Yes, I checked. Both `ctx.BlockHeight()` and `ctx.BlockHeader().Height` have the same value, which seems to be the height of the "current" block being produced.

I guess that's why `LastBlockId.Hash` is prefixed by `Last`.

From [proto/tendermint/types](https://github.com/cometbft/cometbft/blob/main/proto/tendermint/types/types.proto#L54)
![image](https://github.com/pokt-network/poktroll/assets/231488/b7c8dde5-22d2-40c9-b8a9-ea2a8b4d1f91)

The comment seems to indicate that only `last_block_id` is referring to the previous block info (there's no `height` in there)